src/OpenTelemetry.Collector.Dependencies/Implementation/AzureSdkDiagnosticListener.cs

```diff
@@ -61,17 +61,28 @@ public override void OnStartActivity(Activity current, object valueValue)
                 }
             }
 
-            List<Link> links = null;
+            List<Link> parentLinks = null;
             if (LinksPropertyFetcher.Fetch(valueValue) is IEnumerable<Activity> activityLinks)
             {
-                links = new List<Link>();
-                foreach (var link in activityLinks)
+                using (var enumerator = activityLinks.GetEnumerator())
                 {
-                    links.Add(new Link(new SpanContext(link.TraceId, link.ParentSpanId, link.ActivityTraceFlags)));
+                    if (enumerator.MoveNext())
+                    {
+                        parentLinks = new List<Link>();
+                        do
+                        {
+                            var link = enumerator.Current;
+                            if (link != null)
+                            {
+                                parentLinks.Add(new Link(new SpanContext(link.TraceId, link.ParentSpanId, link.ActivityTraceFlags)));
+                            }
+                        }
+                        while (enumerator.MoveNext());
+                    }
                 }
             }
 
-            var span = this.Tracer.StartSpanFromActivity(operationName, Activity.Current, spanKind, links);
+            var span = this.Tracer.StartSpanFromActivity(operationName, Activity.Current, spanKind, parentLinks ?? Array.Empty<Link>() as IEnumerable<Link>);
```

Comment:
can you rather fallback to null? why require it be empty?

Reply:
Just the usual idea that a `null` `IEnumerable<>` is pure evil :)
Also since `Array.Empty<>` doesn't allocate (more than once), I go with that.
I tried to use `null` for the `Func<>` overloads, and `Array.Empty<>` for the `IEnumerable<>` overloads.
Does this make sense or still rather go with null?

Reply:
there are already nulls everywhere in the public API (Start overloads), so why bother with empty array here?
The evil you refer to is null -checking, right? we can check it once or require everyone to write `?? Array.Empty<Link>()`. I prefer to check it once in Span constructor

Reply:
The evil is a joke around twitter. 😄 
@bradwilson https://twitter.com/bradwilson/status/432235119969959936

The reason that`foreach (var a in links)` would throw and usually ppl don't expect null `IEnumerable`s.

Right, for our implementation we'll be checking for null anyway.
Sorry if it sounded pushy, I just wanted to explain the reasoning